### PR TITLE
fix: fix starting lower bound when finding precise access AOS

### PR DESCRIPTION
**Pull request by alex-liang3**
Fixes a bug introduced by https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/583 where, when finding the precise AOS of an access, an earlier access (not found by the coarse search) could be found, leading to the resulting access being more than 1 orbit long. 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved precision of crossing calculations by refining interval boundary handling to avoid incorrect fallback behavior.

* **Tests**
  * Added a new test validating precise crossing detection in edge-case orbital scenarios.

### CODE REVIEW

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp [hunk 1]
```diff
@@ -1215,8 +1215,9 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_7)
         // 2024-12-31 23:57:00 [UTC] | inAccess: 1
         // 2024-12-31 23:58:00 [UTC] | inAccess: 1
         // 2024-12-31 23:59:00 [UTC] | inAccess: 1
+        //     2024-12-31 23:59:30 [UTC] | inAccess: 1
         // 2025-01-01 00:00:00 [UTC] | inAccess: 0
-        // 2025-01-01 00:00:30 [UTC] | inAccess: 1
+        //     2025-01-01 00:00:30 [UTC] | inAccess: 1
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp]: I'm confused about the indentation, what is it supposed to represent here? Can you clarify that in the comments?

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp [hunk 2]
```diff
@@ -1235,6 +1236,86 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_7)
     }
 }
 
+TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_8)
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp]: Nit: May be clearer to rename all of these ComputeAccesses_X more clearly as ComputeAccesses_RegressionTest_X, and start the numbering at 1, since for some reason it starts at 5, idk where 1-4 went?

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp [hunk 3]
```diff
@@ -1235,6 +1236,86 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_7)
...
+                EarthGravitationalModel::EGM2008.J4_,
+                Kepler::PerturbationType::J2
+            ),
+            defaultEarthSPtr_
+        };
+
+        // Create an AccessTarget at the North Pole with an azimuth-elevation mask such that:
+        // - Access 1 (~orbitEpoch - 1 orbit) is longer than the coarse step size
+        // - Access 2 (~orbitEpoch) is shorter than the coarse step size
+        // - Access 3 (~orbitEpoch + 1 orbit) is longer than the coarse step size
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp]:
```suggestion
        // - Access 1 (~orbitEpoch - 1 orbital period) is longer than the coarse step size
        // - Access 2 (~orbitEpoch) is shorter than the coarse step size
        // - Access 3 (~orbitEpoch + 1 orbital period) is longer than the coarse step size
```

**Review by apaletta3 — APPROVED** [review]
Great stuff! Left a few small comments for clarity

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp (hunk 2 above)
Comment (alex-liang3) [reply test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp]: 1-4 are above `ComputeAccesses`, idk why. Can fix in a later PR
